libs/telemetry/src/main/java/org/opensearch/telemetry/tracing/SpanContext.java

```diff
@@ -31,4 +33,17 @@ public SpanContext(Span span) {
     Span getSpan() {
         return span;
     }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
Why do we need `equals` and `hashCode` overrides? Moreover, the `Span` implementation do not implement nor equals nor hashCode

Reply:
This is mostly needed for test cases where mocks are doing equal check.

Reply:
This is public API, we should not implement something that works only for tests

Reply:
Make sense. Let me fix that.